lib/jobs/ScheduleManager.js

```diff
@@ -168,18 +170,32 @@ class ScheduleManager {
 
   static setupSystemJobs() {
     const systemJobDefinition = config.scheduler.system_jobs;
-    _.each(systemJobDefinition, (jobDefinition) => {
+    return Promise.map(systemJobDefinition, (jobDefinition) => {
+      //return added only for UT
       if (jobDefinition.enabled === false) {
         logger.info('Cancelling system job', jobDefinition);
         return this.cancelSchedule(jobDefinition.name, jobDefinition.type);
       }
+      let scheduleSystemJob = false;
       return this
-        .schedule(
-          jobDefinition.name,
-          jobDefinition.type,
-          jobDefinition.interval,
-          jobDefinition.job_data,
-          CONST.SYSTEM_USER);
+        .getSchedule(jobDefinition.name, jobDefinition.type)
+        .then(jobDetails => scheduleSystemJob = jobDetails.repeatInterval !== jobDefinition.interval)
+        .catch(errors.NotFound, () => scheduleSystemJob = true)
```

Comment:
just asking , do we want to catch other run time errors in case DB call returns error ?

Reply:
We want to catch only known errors other runtime errors cant be handled gracefully and its better it be propagated up the chain and let global handler decide what is to be done (i.e. exit and restart etc..)

Reply:
Ok..